### PR TITLE
Add pagesize support to dropdown to limit the height of the dropdown container

### PR DESCRIPTION
**Pull request by timtong1982**
#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [ ] Include a change request file if publishing 
- [ ] New feature, bugfix, or enhancement
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes

(give an overview)

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -189,7 +191,7 @@ export class Dropdown extends BaseComponent<IDropdownInternalProps, IDropdownSta
       onRenderList = this._onRenderList,
       responsiveMode
     } = this.props;
-
+    let { calculatedHeight } = this.state;
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: `const`

**Comment by cschleiden** [comment]
Wouldn't it be easier to host a `List` in the drop down and then provide a way to set `max-height` for the dropdown? The list virtualizes, it can scroll items into view, has means to measure item height, or allows the consumer to pass in own logic. 

What do the others think?

**Comment by micahgodbolt** [comment]
@cschleiden I think we approached this before when I was doing a previous refactor. Would need to ping @dzearing for the specifics again, but I think it had to do with selected items in the virtualized list?

**Comment by micahgodbolt** [comment]
@timtong1982 seems you still have a few linting errors.

**Comment by cschleiden** [comment]
@micahgodbolt @dzearing we probably wouldn't be able to use `FocusZone` (not sure how it deals with virtualize items) but I added support for scrolling items into view to the `List`, maybe that was missing last time?

**Comment by micahgodbolt** [comment]
@cschleiden it's worth looking at again, that's for sure. I agree that a virtualized list would provide support for many more use cases of the dropdown.

File: packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx (hunk 1 above)
Comment (gggjtgxing) [inline packages/office-ui-fabric-react/src/components/Dropdown/Dropdown.tsx]: @cschleiden , I have addressed your comment in another PR: https://github.com/OfficeDev/office-ui-fabric-react/pull/1590

**Comment by gggjtgxing** [comment]
@micahgodbolt @cschleiden , I sent another PR to fix lint errors, please help review: https://github.com/OfficeDev/office-ui-fabric-react/pull/1590

**Comment by gggjtgxing** [comment]
@cschleiden @micahgodbolt, Can we keep the dropdown using div as it is, and open a seperate PR to host a List in dropdown and do other refactor work later?

**Pull request closed without merging**